data/data/bootstrap/files/usr/local/bin/release-image.sh.template

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env bash
+# release-image.sh is a helper library for interacting with a release image.
+
+if ! podman inspect {{.ReleaseImage}} &>/dev/null; then
```

Comment:
If multiple services run this concurrently? what is the behavior?

Reply:
currently `podman` [pulls multiple copies in parallel][1].  containers/image#611 is long-running work around locks to make parallel pulls more efficient, but who knows if/when that will land.

[1]: https://github.com/containers/libpod/issues/1911#issue-386427543

Reply:
Looks like `flock` is available on RHCOS. I can write something off that, unless you have a preference for a different way

Reply:
I was more interested making sure 2 runs of `podman` simultaneously wouldn't cause some corruption.. performance wasn't a concern..

Reply:
I don't think it causes any corruption issues, but if more than a couple things start sourcing this it could get ugly performance-wise.  Is there any objection to using something like `flock`? It's not very portable, though, but it is on RHCOS.  People also do hacky things with `mkdir` as it's guaranteed to be atomic.

```sh
(
    flock 9
    if ! release=$( podman inspect {{.ReleaseImage}} --format '{{"{{"}} index .RepoDigests 0 {{"}}"}}' ) || [[ -z "${release}" ]]; then
        echo "Warning: Could not resolve release image to pull by digest" 2>&1
        release="{{.ReleaseImage}}"
    fi
) 9>/var/lock/release-image.lock
```

Reply:
I'm fine locking somehow in shell.  I'm also fine serializing these two services via some systems magic.  And because this is metal-only, I'm fine not locking at all too ;).  Ahbinav may be more opinionated.

Reply:
You could have a systemd service to pull the release image and have bootkube and ironic depend on it

Reply:
Thanks, @markmc, I think that is a nice approach. I've updated this PR.